var/spack/repos/builtin/packages/qrmumps/package.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack.package import *
+
+
+class Qrmumps(CMakePackage):
+    """a software package for the solution of sparse, linear systems on
+    multicore computers based on the QR factorization of the input matrix."""
+
+    homepage = "https://gitlab.com/qr_mumps/qr_mumps"
+    git = "https://gitlab.com/qr_mumps/qr_mumps"
+    url = "https://gitlab.com/qr_mumps/qr_mumps/-/archive/3.0.4/qr_mumps-3.0.4.tar.gz"
+    maintainers("fpruvost")
+
+    version("master", branch="master")
+    version("3.0.4", sha256="621a294c3bf1e60e4ea6ae29c0586760648947f650e0f86bbabaf82805fc17db")
+
+    variant("amd", default=True, description="Enable AMD ordering")
+    variant("metis", default=True, description="Enable Metis ordering")
+    variant("scotch", default=True, description="Enable Scotch ordering")
+    variant("starpu", default=True, description="Enable StarPU runtime system support")
+    variant("cuda", default=False, description="Enable CUDA")
+    variant("fxt", default=False, description="Enable FxT tracing through StarPU")
+
+    depends_on("pkgconfig", type="build")
+    depends_on("blas")
+    depends_on("lapack")
+    depends_on("suite-sparse", when="+amd")
+    depends_on("metis", when="+metis")
+    depends_on("scotch", when="+scotch")
+    depends_on("starpu", when="+starpu")
+    depends_on("cuda", when="+cuda")
+    depends_on("starpu+cuda", when="+cuda")
```

Comment:
Should the `when` argument be `"+starpu+cuda"`?

Reply:
Yes you're right. I have pushed a new version where cuda and fxt are "conditional variants" depending on the starpu status (should not be exposed if starpu is False).